src/homepageExperience/components/steps/python/WriteDataSql.tsx

```diff
@@ -0,0 +1,283 @@
+import React, {useContext, useEffect, useState} from 'react'
+import {useDispatch, useSelector} from 'react-redux'
+import {
+  Columns,
+  ComponentSize,
+  Grid,
+  InfluxColors,
+  Panel,
+  Table,
+} from '@influxdata/clockface'
+
+import {SafeBlankLink} from 'src/utils/SafeBlankLink'
+import WriteDataHelperBuckets from 'src/writeData/components/WriteDataHelperBuckets'
+import CodeSnippet from 'src/shared/components/CodeSnippet'
+import {WriteDataDetailsContext} from 'src/writeData/components/WriteDataDetailsContext'
+
+import {getOrg} from 'src/organizations/selectors'
+import DataListening from 'src/homepageExperience/components/DataListening'
+import {getBuckets} from 'src/buckets/actions/thunks'
+import {event} from 'src/cloud/utils/reporting'
+
+const logCopyCodeSnippet = () => {
+  event('firstMile.pythonWizard.writeData.code.copied')
+}
+
+const logDocsOpened = () => {
+  event('firstMile.pythonWizard.writeData.docs.opened')
+}
+
+type OwnProps = {
+  onSelectBucket: (bucketName: string) => void
+}
+
+export const WriteDataSqlComponent = (props: OwnProps) => {
+  const org = useSelector(getOrg)
+  const dispatch = useDispatch()
+  const {onSelectBucket} = props
+
+  useEffect(() => {
+    dispatch(getBuckets())
+  }, [dispatch])
```

Comment:
I would only dispatch `getBuckets()` if the `RemoteDataState` of the buckets status is `NotStarted`, so that we don't end up in a situation where we make more than one API call here. Might be okay since the dispatch reference _shouldn't_ change, but better safe than sorry.